### PR TITLE
Add permissions_callback to RocketCDN RESTSubscriber routes

### LINKED ISSUE #2985: PHP Notice: Missing permissions_callback triggered by doing_it_wrong in WP 5.5
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version 👍 
- Used the search feature to ensure that the bug hasn’t been reported before 👍 

**Describe the bug**
After updating to WP 5.5, whenever you open a page for edit or try to create a new post/page a PHP notice is displayed

**Preconditions:**
1. WP Rocket is activated.
2. Site is running WP 5.5

**To Reproduce**
1. Go to the Admin dashboard and click "All pages"
2. Click on "Add New" button

**Expected behavior**
The post editor is displayed.

**Actual Behavior**
The post editor is displayed, but a notice flash for a second, the error is also written to the debug log:
http://snippi.com/s/ekrsai6

**Screenshots**
image

**Backlog Grooming (for WP Media dev team use only)**
- [x] Reproduce the problem
- [x] Identify the root cause
- [x] Scope a solution
- [x] Estimate the effort

### PR DESCRIPTION
**Pull request by iCaspar**
Closes #2985

Adds a `permissions_callback` to REST endpoints.

(Integration tests will need to be updated in the [phpunit](https://github.com/wp-media/phpunit) package for these changes to pass CI.)

### CODE REVIEW

File: inc/Engine/CDN/RocketCDN/RESTSubscriber.php [hunk 1]
```diff
@@ -74,16 +74,17 @@ public function register_enable_route() {
...
 
 							return ! empty( $url );
 						},
-						'sanitize_callback' => function( $param ) {
+						'sanitize_callback' => function ( $param ) {
 							return esc_url_raw( $param );
 						},
 					],
 				],
+				'permission_callback' => __return_true(),
```
Comment (Tabrisrp) [inline inc/Engine/CDN/RocketCDN/RESTSubscriber.php:87]: Looking at the [documentation for permission_callback](https://developer.wordpress.org/rest-api/extending-the-rest-api/adding-custom-endpoints/#permissions-callback), it should be registered like so:

```suggestion
				'permission_callback' => '__return_true',
```

File: inc/Engine/CDN/RocketCDN/RESTSubscriber.php [hunk 2]
```diff
@@ -112,6 +113,7 @@ public function register_disable_route() {
 						'validate_callback' => [ $this, 'validate_key' ],
 					],
 				],
+				'permission_callback' => __return_true(),
```
Comment (Tabrisrp) [inline inc/Engine/CDN/RocketCDN/RESTSubscriber.php:116]:
```suggestion
				'permission_callback' => '__return_true',
```

**Comment by Tabrisrp** [comment]
@wp-media/qa Ready for QA

**Comment by iCaspar** [comment]
With this syntax error fix, all rest-related integration tests are passing.
Failures here are due to remaining 5.5 issues: #2987 and removal of jquery-migrate.

@wp-media/qa Acceptance criteria is that the notice on post edit pages are no longer visible or logged.

**Pull request merged**